File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.cs
public class Solution {
    public IList<string> BuildArray(int[] target, int n) {
        IList<string> Output = new List<string>();
        int currIndex = 0;
        foreach (int num in target){
            while(currIndex+1 <num){
                Output.Add("Push");
                Output.Add("Pop");
                currIndex++;
            }
            Output.Add("Push");
            currIndex++;
        }
        return Output;
    }
}

File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> stackOp = new ArrayList<>();
        int currIndex = 0;
        for (int i = 1; i <= n; i++){
            if (i == target[currIndex]){
                currIndex++;
                stackOp.add("Push");
            }
            else{
                stackOp.add("Push");
                stackOp.add("Pop");
            }
            if (i == target[target.length - 1])
            break;
        }
        return stackOp;

    }
}